kit/Kit.cpp

```diff
@@ -982,6 +984,34 @@ class Document final : public DocumentManagerInterface
         _deltaGen.dropCache();
     }
 
+    void trimIfExcessive()
+    {
+        LOG_TRC("Should we trim our caches ?");
+        if (std::chrono::duration_cast<std::chrono::seconds>(std::chrono::steady_clock::now() -
+                                                             _lastMemTrimTime) < std::chrono::seconds(30))
+        {
+            LOG_TRC("Too soon to trim again");
+            return;
+        }
+
+        double minInactivityMs = std::numeric_limits<double>::max();
+        for (const auto& it : _sessions)
+        {
+            minInactivityMs = std::min(it.second->getInactivityMS(), minInactivityMs);
+        }
+
+        if (minInactivityMs >= 9999)
+        {
+            LOG_DBG("Trimming Core caches");
+            _loKit->trimMemory(4096);
+
+            LOG_DBG("Dropping delta caches");
+            _deltaGen.dropCache();
```

Comment:
Our delta cache saves a ton of CPU and bandwidth, and improves interactivity - we should not drop this easily - please remove this. Possibly we could save some memory there - but the delta cache should be only ~10Mb or so per session I think.